xdg-ninja.sh

```diff
@@ -196,21 +245,21 @@ check_file() {
 # Reads files from programs/, calls check_file on each file specified for each program
 do_check_programs() {
     while IFS="
-" read -r name; read -r filename; read -r movable; read -r help; do
-        check_file "$name" "$filename" "$movable" "$help"
+" read -r name; read -r filename; read -r movable; read -r help; read -r json_file;  do
+        check_file "$name" "$filename" "$movable" "$help" "$json_file"
     done <<EOF
-$(jq 'inputs as $input | $input.files[] as $file | $input.name, $file.path, $file.movable, $file.help' "$(dirname "$0")"/programs/* | sed -e 's/^"//' -e 's/"$//')
+$(jq 'inputs as $input | $input.files[] as $file | $input.name, $file.path, $file.movable, $file.help, input_filename' "$(dirname "$0")"/programs/* | sed -e 's/^"//' -e 's/"$//')
```

Comment:
Why parse json files with slow jq, if all you are going to do is print those?

Reply:
The files need to be parsed by `jq` for `xdg-ninja.sh` to check if the file exists or not.

Reply:
No. Globs don't output non-existent files, you know

Reply:
All files are existent in `./programs/*` that is just a database of the possible files that could exist in ones homedir.

Reply:
> All files are existent in `./programs/*` that is just a database of the possible files that could exist in ones homedir.

Sorry, misunderstood you